data-prepper-plugins/otel-proto-common/src/main/java/com/amazon/dataprepper/plugins/otel/codec/OTelProtoCodec.java

```diff
@@ -66,6 +89,12 @@ public class OTelProtoCodec {
      */
     public static final Function<String, String> SPAN_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> SPAN_ATTRIBUTES + DOT + i.replace(DOT, AT);
     public static final Function<String, String> RESOURCE_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> RESOURCE_ATTRIBUTES + DOT + i.replace(DOT, AT);
+    public static final Function<String, String> PREFIX_AND_METRIC_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> METRIC_ATTRIBUTES + DOT + i.replace(DOT, AT);
+    public static final Function<String, String> PREFIX_AND_RESOURCE_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> RESOURCE_ATTRIBUTES + DOT + i.replace(DOT, AT);
+    public static final Function<String, String> PREFIX_AND_EXEMPLAR_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> EXEMPLAR_ATTRIBUTES + DOT + i.replace(DOT, AT);
+
+
+    private static final Map<Integer, double[]> EXPONENTIAL_BUCKET_BOUNDS = new ConcurrentHashMap<>();
```

Comment:
Yes, it would be possible to introduce memory issues by chance or even malicious intent. Should we perhaps limit the scale to +/- 10 and introduce a config parameter that enables the user to adjust it ? Would 10 be a sensible choice ?

Reply:
@kmssap ,

Yes, using a scale configuration makes sense. And I think 10 is a reasonable default. It will take up 32KB of memory per my calculations.

Reply:
Done